uaclient/event_logger.py

```diff
@@ -103,18 +103,25 @@ def _record_dict_event(
         msg: str,
         service: Optional[str],
         event_dict: List[Dict[str, Optional[str]]],
+        code: Optional[str] = None,
         event_type: Optional[str] = None,
     ):
         if event_type is None:
             event_type = "service" if service else "system"
 
         event_dict.append(
-            {"type": event_type, "service": service, "message": msg}
+            {
+                "type": event_type,
+                "service": service,
+                "message": msg,
+                "message_code": code,
+            }
         )
 
     def error(
         self,
         error_msg: str,
+        error_code: Optional[str] = None,
         service: Optional[str] = None,
         error_type: Optional[str] = None,
     ):
```

Comment:
+1
I think the error codes we will find in status will be similar to the attach ones : invalid / exipred token, connectivity error, etc.
We could add the message codes there in this PR as well, or in a separate one if you think it will be better

Reply:
We could do that, but you also need to modify how our exceptions handle `NamedMessage`, We could store the `NamedMessage` directly on the exception and try to access once it is raised. Do you think that is worth it ?

Reply:
The only problem here is related to exceptions as discussed on the other thread